_pytest/skipping.py

```diff
@@ -230,7 +230,8 @@ def pytest_runtest_makereport(item, call):
     if hasattr(item, '_unexpectedsuccess') and rep.when == "call":
         # we need to translate into how pytest encodes xpass
         rep.wasxfail = "reason: " + repr(item._unexpectedsuccess)
-        rep.outcome = "failed"
+        # TODO: Do we need to check for strict xfail here as well?
```

Comment:
Another thing: I noticed that no tests from `unittest` caught this regression, we should add two new test cases for it:

This test suite should **pass**:

```python
import unittest
class T(unittest.TestCase):

    @unittest.expectedFailure 
    def test_xfail(self):
        assert 0
```

This one should **fail** because `@expectedFailure` works like `xfail(strict=True)`:

```python
import unittest
class T(unittest.TestCase):

    @unittest.expectedFailure 
    def test_unexpected_pass(self):
        pass
```

Additional idea: we could probably parametrize this and run it using `py.test` and `unittest.main` to ensure we have the same outcome in both cases.